presto-main/src/main/java/com/facebook/presto/sql/analyzer/Analysis.java

```diff
@@ -165,137 +165,139 @@ public void setCreateTableAsSelectNoOp(boolean createTableAsSelectNoOp)
 
     public void setAggregates(QuerySpecification node, List<FunctionCall> aggregates)
     {
-        this.aggregates.put(node, aggregates);
+        this.aggregates.put(NodeRef.of(node), aggregates);
     }
 
     public List<FunctionCall> getAggregates(QuerySpecification query)
     {
-        return aggregates.get(query);
+        return aggregates.get(NodeRef.of(query));
     }
 
     public void setOrderByAggregates(OrderBy node, List<Expression> aggregates)
     {
-        this.orderByAggregates.put(node, ImmutableList.copyOf(aggregates));
+        this.orderByAggregates.put(NodeRef.of(node), ImmutableList.copyOf(aggregates));
     }
 
-    public List<Expression> getOrderByAggregates(OrderBy query)
+    public List<Expression> getOrderByAggregates(OrderBy node)
     {
-        return orderByAggregates.get(query);
+        return orderByAggregates.get(NodeRef.of(node));
     }
 
-    public IdentityLinkedHashMap<Expression, Type> getTypes()
+    public Map<NodeRef<Expression>, Type> getTypes()
     {
-        return new IdentityLinkedHashMap<>(types);
+        return types;
```

Comment:
> save this for another PR
sure, but why break defensive copying in the places it's done? Especially for one of the seemingly widely used getter? Consistency is not a satisfactory answer :)